Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -690,35 +905,64 @@ namespace AzFramework
             }
 
             ticket.m_currentRequestId++;
-            return true;
+            return CommandResult::Executed;
+        }
+        else
+        {
+            return CommandResult::Requeue;
+        }
+    }
+
+    auto SpawnableEntitiesManager::ProcessRequest(LoadBarrierCommand& request) -> CommandResult
+    {
+        Ticket& ticket = *request.m_ticket;
+        if (ticket.m_spawnable.IsReady() && request.m_requestId == ticket.m_currentRequestId)
+        {
+            if (request.m_checkAliasSpawnables)
+            {
+                if (Spawnable::EntityAliasConstVisitor visitor = ticket.m_spawnable->TryGetAliasesConst();
+                    !visitor.IsSet() || !visitor.AreAllSpawnablesReady())
+                {
+                    return CommandResult::Requeue;
+                }
+            }
+            
+            request.m_completionCallback(request.m_ticketId);
+            ticket.m_currentRequestId++;
+            return CommandResult::Executed;
         }
         else
         {
-            return false;
+            return CommandResult::Requeue;
         }
     }
 
-    bool SpawnableEntitiesManager::ProcessRequest(DestroyTicketCommand& request)
+    auto SpawnableEntitiesManager::ProcessRequest(DestroyTicketCommand& request) -> CommandResult
     {
         if (request.m_requestId == request.m_ticket->m_currentRequestId)
         {
             for (AZ::Entity* entity : request.m_ticket->m_spawnedEntities)
             {
-                if (entity != nullptr)
+                if (entity != nullptr && !entity->GetComponents().empty())
```

Comment:
Can you explain why you added this? Because an entity can exist without any components. Particularly if users write custom scripts/code to remove all components from entity after spawning. I can see that you are deleting the entity below if components are empty. But deleting the entity is not the same as destroying in the game context since that will send other entity destruction related ebus calls too

Reply:
Looks like you missed answering this. Commenting on this again to get visibility

Reply:
The check was added because entities are not spawned it doesn't have any components. This is because several components depend on entities having at least a transform component. Because those entities are not spawned there's also no reason to destroy them here because it would be referring to non-existing entity (at least from the game context point of view).